lib/colorator.rb

```diff
@@ -13,6 +18,94 @@ module Colorator
     :white   => 37,
     :bold    => 1
   }
+
+  # --------------------------------------------------------------------------
+  # Allows you to check if a string currently has ansi.
+  # --------------------------------------------------------------------------
+
+  def has_ansi?(str)
+    str.match(ANSI_MATCHR).is_a?(
+      MatchData
+    )
+  end
+
+  # --------------------------------------------------------------------------
+  # Jump the cursor, moving it up and then back down to it's spot, allowing
+  # you to do fancy things like multiple output (downloads) the way that Docker
+  # does them in an async way without breaking term.
+  # --------------------------------------------------------------------------
+
+  def ansi_jump(str, num)
+    "\x1b[#{num}A#{clear_line(str)}\x1b[#{
+      num
+    }B"
+  end
+
+  # --------------------------------------------------------------------------
+
+  def reset_ansi(str = "")
+    "\x1b[0m#{
+      str
+    }"
+  end
+
+  # --------------------------------------------------------------------------
+
+  def clear_line(str = "")
+    "\x1b[2K\r#{
+      str
+    }"
+  end
+
+  # --------------------------------------------------------------------------
+  # Strip ANSI from the current string, making it just a normal string.
+  # --------------------------------------------------------------------------
+
+  def strip_ansi(str)
+    str.gsub(
+      ANSI_MATCHR, ""
+    )
+  end
+
+  # --------------------------------------------------------------------------
+  # Clear the screen's current view, so the user gets a clean output.
+  # --------------------------------------------------------------------------
+
+  def clear_screen(str = "")
+    "\x1b[H\x1b[2J#{
+      str
+    }"
+  end
+
+  # --------------------------------------------------------------------------
+
+  def colorize(str = "", color)
+    "\x1b[#{color}m#{str}\x1b[0m"
```

Comment:
[I only see `\e` on Wikipedia](https://en.wikipedia.org/wiki/ASCII). Can you point me to documentation for `\x1b`? I may as well learn while we go here.